experimental/PlaneCurve/ProjEllipticCurve.jl

```diff
@@ -0,0 +1,459 @@
+export ProjEllipticCurve, discriminant, issmooth, j_invariant,
+       Point_EllCurve, curve, weierstrass_form, toweierstrass,
+       iselliptic
+
+################################################################################
+# Helping functions
+################################################################################
+
+function isweierstrassform(eq::Oscar.MPolyElem{S}) where S <: FieldElem
+   R = parent(eq)
+   x = gen(R, 1)
+   y = gen(R, 2)
+   z = gen(R, 3)
+   A = [x*y*z, y*z^2, x^2*z, x*z^2, z^3]
+   f = eq - y^2*z + x^3
+   for t in A
+      f = f - coeff(eq, t)*t
+   end
+   return iszero(f)
+end
+
+################################################################################
+
+function shortformtest(eq::Oscar.MPolyElem{S}) where S <: FieldElem
+   R = parent(eq)
+   x = gen(R, 1)
+   y = gen(R, 2)
+   z = gen(R, 3)
+   A = [coeff(eq, x*y*z), -coeff(eq, x^2*z), coeff(eq, y*z^2), -coeff(eq, x*z^2), -coeff(eq, z^3)]
+   iszero(A[1]) && iszero(A[2]) && iszero(A[3]) && return (true, [A[4], A[5]])
+   return (false, A)
+end
+
+################################################################################
+
+function iselliptic(F::Oscar.MPolyElem_dec)
```

Comment:
I think "elliptic polynomial" has a different meaning. How about not defining this method? Is it very important?

Reply:
Maybe just call it `_iselliptic`, so that the method is not exported together with the useful one below.